tools/setup_helpers/cudnn.py

```diff
@@ -34,6 +32,9 @@
         'C_INCLUDE_PATH',
         'CPLUS_INCLUDE_PATH',
     ])))
+    # Add CUDA related dirs to candidate list
+    for suffix in LIB_SUFFIXS:
+        lib_paths.append(os.path.join(CUDA_HOME, suffix))
```

Comment:
the ordering in lib_paths matters. You've appended `CUDA_HOME + suffix` entries to `lib_paths`, which means they will be considered after system paths like `/usr/lib/x86_64-linux-gnu/`. What you need to do is insert them right after os.getenv('CUDNN_LIB_DIR')